### ISSUE TITLE
tmux flat out not working

### ISSUE THREAD

**Issue by paanvaannd** [tmux/tmux#263]
Running OS X 10.11 (El Capitan) completely up-to-date and using iTerm2. I was using tmux and it was working like a charm until yesterday, and since then tmux will not even start. I use zsh as my shell and tried running it from there, to no avail. Running from bash has no effect either. I've cloned the git repo and built it using the commands given in the README, including installing `automake` and `libgen` (up-to-date as well). I initially had it installed through Homebrew but now installing through Homebrew or building it using the instructions listed in the README do not work. There is no output given after running the command, it simply does not work:
<img width="1920" alt="tmux-issue" src="https://cloud.githubusercontent.com/assets/16567134/12270980/87fe4304-b91e-11e5-878b-029c4e6c7696.png">

Since the time it was working and now, the only thing I had done was messed with `zsh` files, including installing `antigen`. There was a [tmux plugin](https://github.com/thewtex/tmux-mem-cpu-load) that I installed recently through antigen, and that is the last thing I really messed with concerning tmux in any way. In the install instructions there, it mentions that I should include some code within the .tmux.conf file to properly use the plugin, and forgetting to do that, I tried to start tmux right off the bat which is when tmux first crashed on me, and has been happening since. Of course, I don't think that forgetting to include 3 lines of code within the tmux configuration file would entirely screw up my settings to where tmux just won't work at all, but I've included this information since it is the last thing I did before tmux stopped working for me.

Any suggestions as to any settings I should change, files to install/delete/move, etc.?

Edit: completed incomplete issue report.

**Comment by nicm** [comment]
What if you do:

tmux -Ltest -f/dev/null new

**Comment by paanvaannd** [comment]
Still nothing, unfortunately:
<img width="1920" alt="tmux-ltest" src="https://cloud.githubusercontent.com/assets/16567134/12295943/08ac3e74-b9ca-11e5-92ad-aad26cf5f43e.png">

**Comment by nicm** [comment]
Run:

pkill tmux
tmux -Ltest -f/dev/null -vvvv new

Then show me the tmux-server*.log file.

**Comment by paanvaannd** [comment]
[tmux-server-98268_log.txt](https://github.com/tmux/tmux/files/88923/tmux-server-98268_log.txt)

**Comment by nicm** [comment]
Looks like your shell is just exiting when you start it in tmux, try setting default-shell to /bin/sh and if that works then check your profile files.

**Comment by paanvaannd** [comment]
I've set my default shell to /bin/sh and tried running tmux from that, but it still exits as soon as it starts. I haven't yet checked my profile files, since I don't know what those are. Sorry, I'm moderately new to using the command line and to programming in general so I'm not sure what the profile files are (or if I do know, I just haven't heard that term before).

**Comment by nicm** [comment]
Try this:

mv .tmux.conf .tmux.conf.old
echo "set -g default-shell /bin/sh" >.tmux.conf
pkill tmux
tmux -vvvv new

And does it work?

If not, please show me the new tmux-server*.log

**Comment by paanvaannd** [comment]
Ah okay, sorry I thought you meant to set the default system shell to sh instead of zsh. Modified the appropriate files and tried running it again but still nothing. Here's the new log:
[tmux-server-98843_log.txt](https://github.com/tmux/tmux/files/88981/tmux-server-98843_log.txt)

**Comment by nicm** [comment]
Hmm weird.

Do this please:

pkill tmux
strace -fv -ostrace.out tmux new

Then show me the strace.out log.

**Comment by nicm** [comment]
Hmm maybe no strace on OS X...

**Comment by paanvaannd** [comment]
Would `dtruss` work?
https://www.quora.com/Whats-the-equivalent-of-strace-in-OS-X?share=1

**Comment by nicm** [comment]
Sure but you will need to figure out how to use it yourself.

**Comment by paanvaannd** [comment]
Already working on it! Trying to figure out what the `-fv` command appended to `strace` is so I can find a `dtruss` equivalent. Could you let me know what you're trying to accomplish using the command you gave me so I can try to find an approximation using dtruss instead?

Also, I will be busy for a couple hours so it may take a while for me to respond/figure things out but I'll try to get things done when possible.

**Comment by nicm** [comment]
You need to make it follow child processes (-f). Don't worry about -v.

**Comment by paanvaannd** [comment]
Ah okay! `dtruss -f` (according to the man page, `-f` follows child processes as well) gives the following:
<img width="419" alt="dtruss" src="https://cloud.githubusercontent.com/assets/16567134/12297515/21ebd2e2-b9d3-11e5-835e-3811b8bf808d.png">
Note: I ran the command twice initially (pictured) and the first time it output a list of 2 processes while the second time it only output 1. All other times I ran it gave 2 outputs, if that is of any assistance.

**Comment by nicm** [comment]
I don't think -o does the same thing.

**Comment by paanvaannd** [comment]
Okay, so you're trying to write outputs to a file called `strace.out`, got it! I'll look up the relevant equivalent in `dtruss` and get back to you in a couple hours. Thank you for your assistance so far!

**Comment by paanvaannd** [comment]
Okay, so you're trying to write outputs to a file called `strace.out`, got it! I'll look up the relevant equivalent in `dtruss` and get back to you in a couple hours. Thank you for your assistance so far!

**Comment by paanvaannd** [comment]
Sorry for the closing and reopening, I hit the wrong button. I've figured out how to output `dtruss` results to a file (attached):
[dtruss.txt](https://github.com/tmux/tmux/files/89251/dtruss.txt)

**Comment by nicm** [comment]
Looks like there is something wrong with /dev/ptmx on your system.

**Comment by paanvaannd** [comment]
Could you let me know, generally or specifically, what is wrong with my /dev/ptmx so that I could look up how to remedy that? I ran a bunch of disk utility processes to see if that would fix any permissions or errors if that's the problem but that hasn't fixed tmux yet. Any recommendations as to what I should do, or where I should go to in order to fix this?

**Comment by paanvaannd** [comment]
So it's working now... I did nothing to fix it, I don't think.

I restarted my computer and booted into recovery mode, ran a disk utility application there, and logged in again and now it's working, though I had done this same thing previously and the  utility application always claimed there was never any disk corruption. I did do this in a tech store, however, and I talked about doing a clean install on my computer while trying to launch tmux just one last time before doing that, so maybe that scared the computer enough to make it behave? In any case now, it's working fine with both `sh` and `zsh`.

Thank you for your assistance! I'll close this issue now and open it again should I encounter any other problems and find that those issues can be replicated.
